notes/stv/20191120-openstack-deployment.txt

```diff
@@ -0,0 +1,761 @@
+
+spark.master                     yarn
+spark.driver.memory              40g
+spark.yarn.am.memory            40g
+spark.executor.memory          40g
+spark.eventLog.enabled  true
+spark.eventLog.dir hdfs://${stv-aglais-master-ip:?}:9000/spark-log
+
+
+2019-11-20 19:04:39,482 INFO scheduler.DAGScheduler: Job 0 finished: reduce at SparkPi.scala:38, took 0.706995 s
+Pi is roughly 3.1443231443231445
+
+
+
+
+#-------------------------------------------------------------------------
+# Zeppelin
+#-------------------------------------------------------------------------
+
+
+
+## On Zeppelin Node: (ssh stv-aglais-zeppelin)
+
+## Fetch Zeppelin Binaries
+sudo yum install wget
+sudo yum install nano
+sudo yum install java-1.8.0-openjdk
+
+wget https://www-eu.apache.org/dist/zeppelin/zeppelin-0.8.2/zeppelin-0.8.2-bin-all.tgz
+tar -xzvf zeppelin-0.8.2-bin-all.tgz 
+mv zeppelin-0.8.2-bin-all zeppelin
+rm zeppelin-0.8.2-bin-all.tgz 
+
+
+
+## Setup Zeppelin Configurations
+
+## Setup Users in shiro.ini
+cp /home/fedora/zeppelin/conf/shiro.ini.template /home/fedora/zeppelin/conf/shiro.ini
+
+
+## Set Zeppelin Configuration
+cp /home/fedora/zeppelin/conf/zeppelin-site.xml.template /home/fedora/zeppelin/conf/zeppelin-site.xml
+nano /home/fedora/zeppelin/conf/zeppelin-site.xml
+
+
+
+sudo su
+cat <<EOF >> "/etc/hosts"
+
+127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
+::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
+
+	
+10.0.0.15       stv-aglais-master.novalocal
+10.0.0.15       stv-aglais-master
+10.0.0.16       stv-aglais-worker-1
+10.0.0.9       stv-aglais-worker-2
+10.0.0.10       stv-aglais-worker-3
+10.0.0.5      stv-aglais-zeppelin
+
+EOF
+  
+exit
+
+
+cat > "${HOME:?}/.ssh/config" << EOF
+
+    Host stv-aglais-worker-1
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-worker-2 
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-worker-3
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-master.novalocal
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-master
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+    Host stv-aglais-zeppelin
+        User fedora
+        IdentityFile ~/.ssh/stv-master
+        Protocol 2
+        ForwardAgent yes
+        PasswordAuthentication no
+
+EOF
+
+
+sudo chmod 600 ~/.ssh/config
+
+
+
+mkdir /home/fedora/spark
+
+## Copy Hadoop & Spark Config files from master
+ssh stv-aglais-master
```

Comment:
Why login to `stv-aglais-master` and push rather than just pull the files we want ?